qiskit/chemistry/components/variational_forms/uccsd.py

```diff
@@ -419,10 +419,15 @@ def construct_circuit(self, parameters, q=None):
                                task_args=(q, self._num_time_slices),
                                num_processes=aqua_globals.num_processes)
 
-        for qc in results:
-            if self._shallow_circuit_concat:
-                circuit.data += qc.data
-            else:
+        if self._shallow_circuit_concat:
+            for qc in results:
+                for _, qbits, _ in qc._data:
+                    for qbit in qbits:
+                        qbit._register = q
```

Comment:
Updating `_register` could lead to some inconsistent dangling references that could lead to some tricky bugs down the line. (e.g. `Bit`s hold a reference to their `Register`, and `Register`s also hold a reference to all their `Bit`s. ) Can you use a mapping into `circuit.qubits` to replace the `Qubit` instances in `qbits`?